test/auth_state_test.dart

```diff
@@ -0,0 +1,318 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:kinde_flutter_sdk/src/token/auth_state.dart';
+import 'package:kinde_flutter_sdk/src/model/token_type.dart';
+
+import 'test_helpers/jwt_test_helper.dart';
+import 'test_helpers/auth_state_fixtures.dart';
+
+void main() {
+  group('AuthState', () {
+    group('Constructor', () {
+      test('should create AuthState with all parameters', () {
+        final expiresAt = DateTime.now().add(const Duration(hours: 1));
+        final state = AuthState(
+          accessToken: 'access_token',
+          refreshToken: 'refresh_token',
+          idToken: 'id_token',
+          accessTokenExpirationDateTime: expiresAt,
+          grantType: 'authorization_code',
+          scope: 'openid profile',
+        );
+
+        expect(state.accessToken, 'access_token');
+        expect(state.refreshToken, 'refresh_token');
+        expect(state.idToken, 'id_token');
+        expect(state.accessTokenExpirationDateTime, expiresAt);
+        expect(state.grantType, 'authorization_code');
+        expect(state.scope, 'openid profile');
+      });
+
+      test('should default grantType to refresh_token', () {
+        const state = AuthState();
+        expect(state.grantType, 'refresh_token');
+      });
+
+      test('should allow null values for optional fields', () {
+        const state = AuthState();
+        expect(state.accessToken, isNull);
+        expect(state.refreshToken, isNull);
+        expect(state.idToken, isNull);
+        expect(state.accessTokenExpirationDateTime, isNull);
+        expect(state.scope, isNull);
+      });
+    });
+
+    group('fromJson', () {
+      test('should parse valid JSON with expires_in', () {
+        final json = AuthStateFixtures.createValidJson(expiresIn: 3600);
+        final state = AuthState.fromJson(json);
+
+        expect(state.accessToken, isNotNull);
+        expect(state.refreshToken, 'test_refresh_token');
+        expect(state.idToken, isNotNull);
+        expect(state.grantType, 'refresh_token');
+        expect(state.scope, 'openid profile email offline');
+        expect(state.accessTokenExpirationDateTime, isNotNull);
+
+        // Expiry should be approximately now + 3600 seconds
+        final expectedExpiry = DateTime.now().add(const Duration(seconds: 3600));
+        expect(
+          state.accessTokenExpirationDateTime!.difference(expectedExpiry).inSeconds.abs(),
+          lessThan(5), // Allow 5 second tolerance
+        );
+      });
+
+      test('should parse JSON with ISO date string', () {
+        final expectedExpiry = DateTime.now().add(const Duration(hours: 2));
+        final json = AuthStateFixtures.createJsonWithIsoDate(expiresAt: expectedExpiry);
+        final state = AuthState.fromJson(json);
+
+        expect(state.accessTokenExpirationDateTime, isNotNull);
+        expect(
+          state.accessTokenExpirationDateTime!.difference(expectedExpiry).inSeconds.abs(),
+          lessThan(1),
+        );
+      });
```

Comment:
See https://github.com/kinde-oss/kinde-flutter-sdk/pull/53/files#r2573210667. 


```suggestion
      test('should parse JSON with ISO date string', () {
        final testDateTime = DateTime(2025, 1, 1);
        withClock(Clock.fixed(testDateTime), () {
          final expectedExpiry = clock.now().add(const Duration(hours: 2));
          final json = AuthStateFixtures.createJsonWithIsoDate(
            expiresAt: expectedExpiry,
          );
          final state = AuthState.fromJson(json);

          expect(state.accessTokenExpirationDateTime, isNotNull);
          expect(state.accessTokenExpirationDateTime!, expectedExpiry);
        });
      });
```